src/lib/db/group-store.ts

```diff
@@ -61,10 +61,14 @@ const groupToRow = (group: IStoreGroup) => ({
     mappings_sso: JSON.stringify(group.mappingsSSO),
 });
 
-export default class GroupStore implements IGroupStore {
-    private db: Knex;
+export default class GroupStore
+    extends Transactor<GroupStore>
```

Comment:
is it possible to use [Mixins](https://www.typescriptlang.org/docs/handbook/mixins.html) instead of inheritance to provide the transactional capabilities to the store?

Reply:
This is actually what I tried first, because I would rather avoid inheritance if possible. Unfortunately, you can't hook into the constructor type from a mixin so the cloning approach here doesn't work. If anyone has a suggestion for making that work I would be a ton happier with a mixin though